docs/modules/ROOT/pages/governance.adoc

```diff
@@ -319,3 +320,105 @@ await governor.execute(
 ```
 
 Executing the proposal will transfer the ERC20 tokens to the chosen recipient. To wrap up: we set up a system where a treasury is controlled by the collective decision of the token holders of a project, and all actions are executed via proposals enforced by on-chain votes.
+
+== Timestamp based governance
+
+=== Motivation
+
+It is sometimes difficult for humans to deal with durations expressed in number of blocks because of inconstistent or unpredictable time between blocks. That is particularly true of some L2 networks where blocks are being produced based on blockchain usage. Using number of blocks can also lead to the governance rules being affected by network upgrades eventually modifying the mean time between blocks.
+
+The difficulty of replacing block numbers with timestamp is that the governor and the token must both use the same format when querying past votes. If a token is designed around block numbers, it is not possible for a governor to reliably do timestamp based lookups.
+
+Therefore, designing a timestamp based voting system starts with the token.
+
+=== Token
+
+Since v4.9, all voting contracts (including xref:api:token/ERC20.adoc#ERC20Votes[`ERC20Votes`] and xref:api:token/ERC721.adoc#ERC721Votes[`ERC721Votes`]) rely on xref:api:interfaces.adoc#IERC6372[IERC6372] for clock management. In order to change from operating with block numbers to operating with timestamps, all that is required is to override the `clock()` function. When doing so, one should not forget to also update the `CLOCK_MODE()` function.
```

Comment:
```suggestion
Since v4.9, all voting contracts (including xref:api:token/ERC20.adoc#ERC20Votes[`ERC20Votes`] and xref:api:token/ERC721.adoc#ERC721Votes[`ERC721Votes`]) rely on xref:api:interfaces.adoc#IERC6372[IERC6372] for clock management. In order to change from operating with block numbers to operating with timestamps, all that is required is to override the `clock()` and `CLOCK_MODE()` functions.
```